Tests/TestingApps/PaywallsTester/PaywallsTester/UI/Views/OfferingList/ManagePaywallButton.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  ManagePaywallButton.swift
+//  PaywallsTester
+//
+//  Created by James Borthwick on 2024-04-30.
+//
+
+import SwiftUI
+
+struct ManagePaywallButton: View {
+
+    enum Kind: String {
+        case edit
+        case new
+
+        var systemImageName: String {
+            switch self {
+            case .edit:
+                return "slider.horizontal.2.square.on.square"
+            case .new:
+                return "plus.square.on.square"
+            }
+        }
+    }
+
+    var body: some View {
+        Button {
+            guard let url = dashboardPaywallURL else { return }
+            openURL(url)
+        } label: {
+            switch kind {
+            case .edit:
+                HStack {
+                    Text(buttonName ?? "Edit Paywall")
+                        .font(.headline)
+                    Spacer()
+                    Image(systemName: kind.systemImageName)
+                }
+            case .new:
+                HStack {
+                    Text(buttonName ?? "New Paywall")
+                        .font(.headline)
+                    Spacer()
+                    Image(systemName: "escape")
+                        .rotationEffect(Angle(degrees: 90))
```

Comment:
should this be `Image(systemName: kind.systemImageName)` instead? 
If so, you could unify the name into a var (see previous suggestion) and then remove the `switch` here, where the same button works for both kinds

Reply:
This button in "edit" mode is used in two locations, on the main screen for Offerings without paywalls, and also in the Menu that you get by tapping the ellipsis on the Offerings with Paywalls. It uses a different icon in each location. "escape" is used in the Offerings list and I like it best because it communicates that it is going to exit the app. Unfortunately the `.rotationEffect` modifier has not effect when used on an Image in Menu, and without rotating it the icon points up and to the left, which looks wrong. So I kept it for the Offerings list, where I can rotate it, and used a different icon in the Menu, and coding it as the "standard" icon because it will work everywhere.

The best thing to do here (imo) would be to create a custom escape icon that is pre-rotated that we can use in both locations. But I didn't think it was worth the time at this point.

Thoughts?

Reply:
I'd agree that it's not worth the effort. It did feel pretty awkward to me to see a `Kind` enum that includes a system image and then it not being used for 1 of the 2 possible values

Reply:
maybe it should be a third kind?

Reply:
Actually some of what I said was outdated! Reworked this, much better, no more switching the view body. https://github.com/RevenueCat/purchases-ios/pull/3856/commits/9ee4f800633bdca325d5a47f9479e04bee76e2d7